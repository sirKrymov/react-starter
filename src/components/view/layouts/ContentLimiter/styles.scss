@import '../../../../styles/mixins';

.content-limiter {
  max-width: 1704px;
  margin: 0 auto;
  height: 100%;
  width: 100%;

  &--fluid {
    max-width: 100%;
  }
}
