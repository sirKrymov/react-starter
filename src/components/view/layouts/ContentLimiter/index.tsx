// @SETUP: Please don't forget to remove this component or current comment if component is used in project
// @TODO: Add component tests
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  children: ReactNode;
  fluid?: boolean;
}

export function ContentLimiter({
  className,
  children,
  fluid
}: IProps): ReactElement {
  const contentLimiterTemplateClass = cn(
    {
      'content-limiter--fluid': fluid,
      'content-limiter': true
    },
    className
  );

  return <div className={contentLimiterTemplateClass}>{children}</div>;
}
