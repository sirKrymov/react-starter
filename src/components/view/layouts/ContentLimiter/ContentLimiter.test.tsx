import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ContentLimiter } from '../ContentLimiter';

describe('<ContentLimiter> layout component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = (): void => {
    shallowComponent = shallow(
      <ContentLimiter>
        <div>Test component</div>
      </ContentLimiter>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
