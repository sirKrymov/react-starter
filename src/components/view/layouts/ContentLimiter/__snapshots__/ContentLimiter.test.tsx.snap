// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentLimiter> layout component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="content-limiter"
>
  <div>
    Test component
  </div>
</div>
`;
